Tests fail with Go1.12rc1

With Go 1.12rc1 when I run `go test go.uber.org/zap` I see the following.  I think this is due to the changes in https://golang.org/issue/26839.

```
invalid value "unknown" for flag -level: unrecognized level: "unknown"
Usage of test:
  -level value
    	minimum enabled logging level
--- FAIL: TestNewStdLog (0.00s)
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:101
        	            				common_test.go:40
        	            				global_test.go:98
        	Error:      	Expect "/home/iant/gopath/src/go.uber.org/zap/common_test.go:40" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLog
        	Messages:   	Unexpected caller annotation.
--- FAIL: TestNewStdLogAt (0.00s)
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:113
        	            				common_test.go:40
        	            				global_test.go:109
        	Error:      	Expect "/home/iant/gopath/src/go.uber.org/zap/common_test.go:40" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLogAt
        	Messages:   	Unexpected caller annotation.
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:113
        	            				common_test.go:40
        	            				global_test.go:109
        	Error:      	Expect "/home/iant/gopath/src/go.uber.org/zap/common_test.go:40" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLogAt
        	Messages:   	Unexpected caller annotation.
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:113
        	            				common_test.go:40
        	            				global_test.go:109
        	Error:      	Expect "/home/iant/gopath/src/go.uber.org/zap/common_test.go:40" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLogAt
        	Messages:   	Unexpected caller annotation.
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:113
        	            				common_test.go:40
        	            				global_test.go:109
        	Error:      	Expect "/home/iant/gopath/src/go.uber.org/zap/common_test.go:40" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLogAt
        	Messages:   	Unexpected caller annotation.
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:113
        	            				common_test.go:40
        	            				global_test.go:109
        	Error:      	Expect "/home/iant/gopath/src/go.uber.org/zap/common_test.go:40" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLogAt
        	Messages:   	Unexpected caller annotation.
--- FAIL: TestNewStdLogAtPanics (0.00s)
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:126
        	            				common_test.go:40
        	            				global_test.go:122
        	Error:      	Expect "/home/iant/gopath/src/github.com/stretchr/testify/assert/assertions.go:862" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLogAtPanics
        	Messages:   	Unexpected caller annotation.
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:126
        	            				common_test.go:40
        	            				global_test.go:122
        	Error:      	Expect "/home/iant/gopath/src/github.com/stretchr/testify/assert/assertions.go:862" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLogAtPanics
        	Messages:   	Unexpected caller annotation.
--- FAIL: TestNewStdLogAtFatal (0.00s)
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:137
        	            				exit.go:53
        	            				global_test.go:133
        	            				common_test.go:40
        	            				global_test.go:132
        	Error:      	Expect "/home/iant/gopath/src/go.uber.org/zap/internal/exit/exit.go:53" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestNewStdLogAtFatal
        	Messages:   	Unexpected caller annotation.
--- FAIL: TestRedirectStdLogCaller (0.00s)
    global_test.go:174: 
        	Error Trace:	global_test.go:174
        	            				common_test.go:40
        	            				global_test.go:169
        	Error:      	"/home/iant/gopath/src/go.uber.org/zap/common_test.go" does not contain "global_test.go"
        	Test:       	TestRedirectStdLogCaller
        	Messages:   	Unexpected caller annotation.
--- FAIL: TestRedirectStdLogAtCaller (0.00s)
    global_test.go:213: 
        	Error Trace:	global_test.go:213
        	            				common_test.go:40
        	            				global_test.go:206
        	Error:      	"/home/iant/gopath/src/go.uber.org/zap/common_test.go" does not contain "global_test.go"
        	Test:       	TestRedirectStdLogAtCaller
        	Messages:   	Unexpected caller annotation.
    global_test.go:213: 
        	Error Trace:	global_test.go:213
        	            				common_test.go:40
        	            				global_test.go:206
        	Error:      	"/home/iant/gopath/src/go.uber.org/zap/common_test.go" does not contain "global_test.go"
        	Test:       	TestRedirectStdLogAtCaller
        	Messages:   	Unexpected caller annotation.
    global_test.go:213: 
        	Error Trace:	global_test.go:213
        	            				common_test.go:40
        	            				global_test.go:206
        	Error:      	"/home/iant/gopath/src/go.uber.org/zap/common_test.go" does not contain "global_test.go"
        	Test:       	TestRedirectStdLogAtCaller
        	Messages:   	Unexpected caller annotation.
    global_test.go:213: 
        	Error Trace:	global_test.go:213
        	            				common_test.go:40
        	            				global_test.go:206
        	Error:      	"/home/iant/gopath/src/go.uber.org/zap/common_test.go" does not contain "global_test.go"
        	Test:       	TestRedirectStdLogAtCaller
        	Messages:   	Unexpected caller annotation.
    global_test.go:213: 
        	Error Trace:	global_test.go:213
        	            				common_test.go:40
        	            				global_test.go:206
        	Error:      	"/home/iant/gopath/src/go.uber.org/zap/common_test.go" does not contain "global_test.go"
        	Test:       	TestRedirectStdLogAtCaller
        	Messages:   	Unexpected caller annotation.
--- FAIL: TestRedirectStdLogAtPanics (0.00s)
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:230
        	            				common_test.go:40
        	            				global_test.go:225
        	Error:      	Expect "/home/iant/gopath/src/github.com/stretchr/testify/assert/assertions.go:862" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestRedirectStdLogAtPanics
        	Messages:   	Unexpected caller annotation.
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:230
        	            				common_test.go:40
        	            				global_test.go:225
        	Error:      	Expect "/home/iant/gopath/src/github.com/stretchr/testify/assert/assertions.go:862" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestRedirectStdLogAtPanics
        	Messages:   	Unexpected caller annotation.
--- FAIL: TestRedirectStdLogAtFatal (0.00s)
    global_test.go:274: 
        	Error Trace:	global_test.go:274
        	            				global_test.go:248
        	            				exit.go:53
        	            				global_test.go:243
        	            				common_test.go:40
        	            				global_test.go:242
        	Error:      	Expect "/home/iant/gopath/src/go.uber.org/zap/internal/exit/exit.go:53" to match "go.uber.org/zap/global_test.go:\d+$"
        	Test:       	TestRedirectStdLogAtFatal
        	Messages:   	Unexpected caller annotation.
--- FAIL: TestStacktraceFiltersVendorZap (0.00s)
    require.go:794: 
        	Error Trace:	stacktrace_ext_test.go:106
        	            				stacktrace_ext_test.go:144
        	            				stacktrace_ext_test.go:89
        	Error:      	Received unexpected error:
        	            	stat /home/iant/gopath/src/go.uber.org/zap/vendor/github.com/stretchr/testify: no such file or directory
        	Test:       	TestStacktraceFiltersVendorZap
        	Messages:   	Required dependency (/home/iant/gopath/src/go.uber.org/zap/vendor/github.com/stretchr/testify) not installed in vendor
FAIL
FAIL	go.uber.org/zap	0.196s
```

--- a/global.go
+++ b/global.go
@@ -31,7 +31,6 @@ import (
 )
 
 const (
-	_stdLogDefaultDepth      = 2
 	_loggerWriterDepth       = 2
 	_programmerErrorTemplate = "You've found a bug in zap! Please file a bug at " +
 		"https://github.com/uber-go/zap/issues/new and reference this error: %v"

--- a/global_go112.go
+++ b/global_go112.go
@@ -1,0 +1,26 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// See #682 for more information.
+// +build go1.12
+
+package zap
+
+const _stdLogDefaultDepth = 1

--- a/global_prego112.go
+++ b/global_prego112.go
@@ -1,0 +1,26 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// See #682 for more information.
+// +build !go1.12
+
+package zap
+
+const _stdLogDefaultDepth = 2

